src/overlays/actors/ovl_En_Ishi/z_en_ishi.c

```diff
@@ -21,47 +55,420 @@ const ActorInit En_Ishi_InitVars = {
+    }
+}
+
+void EnIshi_Fall(EnIshi* this) {
+    this->actor.velocity.y += this->actor.gravity;
+
+    if (this->actor.velocity.y < this->actor.minVelocityY) {
+        this->actor.velocity.y = this->actor.minVelocityY;
+    }
+}
+
+void func_80A7ED94(Vec3f* arg0, f32 arg1) {
+    arg1 += ((Math_Rand_ZeroOne() * 0.2f) - 0.1f) * arg1;
+    arg0->x -= arg0->x * arg1;
+    arg0->y -= arg0->y * arg1;
+    arg0->z -= arg0->z * arg1;
+}
+
+void EnIshi_SpawnBugs(EnIshi* this, GlobalContext* globalCtx) {
+    s32 i;
+
+    for (i = 0; i < 3; i++) {
+        if (Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_INSECT, this->actor.posRot.pos.x,
+                        this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, Math_Rand_ZeroOne() * 0xFFFF, 0,
+                        1) == NULL) {
+            break;
+        }
+    }
+}
+
+static InitChainEntry sInitChains[][5] = {
+    {
+        ICHAIN_F32_DIV1000(gravity, 64336, ICHAIN_CONTINUE),
+        ICHAIN_F32_DIV1000(minVelocityY, 45536, ICHAIN_CONTINUE),
+        ICHAIN_F32(uncullZoneForward, 1200, ICHAIN_CONTINUE),
+        ICHAIN_F32(uncullZoneScale, 150, ICHAIN_CONTINUE),
+        ICHAIN_F32(uncullZoneDownward, 400, ICHAIN_STOP),
+    },
+    {
+        ICHAIN_F32_DIV1000(gravity, 63036, ICHAIN_CONTINUE),
+        ICHAIN_F32_DIV1000(minVelocityY, 45536, ICHAIN_CONTINUE),
+        ICHAIN_F32(uncullZoneForward, 2000, ICHAIN_CONTINUE),
+        ICHAIN_F32(uncullZoneScale, 250, ICHAIN_CONTINUE),
+        ICHAIN_F32(uncullZoneDownward, 500, ICHAIN_STOP),
+    },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ishi/func_80A7E5A8.s")
+void EnIshi_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnIshi* this = THIS;
+    s16 type = this->actor.params & 1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ishi/func_80A7E824.s")
+    Actor_ProcessInitChain(&this->actor, sInitChains[type]);
+    if (globalCtx->csCtx.state != 0) {
+        this->actor.uncullZoneForward += 1000.0f;
+    }
+    if (this->actor.shape.rot.y == 0) {
+        this->actor.shape.rot.y = this->actor.posRot.rot.y = Math_Rand_ZeroFloat(0x10000);
+    }
+    Actor_SetScale(&this->actor, sRockScales[type]);
+    EnIshi_InitCollider(&this->actor, globalCtx);
+    if ((type == ROCK_LARGE) &&
+        Flags_GetSwitch(globalCtx, ((this->actor.params >> 0xA) & 0x3C) | ((this->actor.params >> 6) & 3))) {
+        Actor_Kill(&this->actor);
+    } else {
+        func_80061ED4(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+        this->actor.shape.unk_08 = D_80A7FA20[type];
+        if (!((this->actor.params >> 5) & 1) && !EnIshi_SnapToFloor(this, globalCtx, 0.0f)) {
+            Actor_Kill(&this->actor);
+        } else {
+            EnIshi_SetupWait(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ishi/func_80A7EB10.s")
+void EnIshi_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    Collider_DestroyCylinder(globalCtx, &THIS->collider);
```

Comment:
think we talked about this in discord already, but a this temp doesnt work right?

Reply:
Yep, having a temp causes regalloc.